File: result/translate/starchat/java2python/python/NEXT_HIGHER_NUMBER_WITH_SAME_NUMBER_OF_SET_BITS.py
def f_gold(x):
        if x > 0:
            rightOne, nextHigherOneBit, rightOnesPattern, next = 0, 0, 0, 0
            rightOne = x & -x
            nextHigherOneBit = x + rightOne
            rightOnesPattern = x ^ nextHigherOneBit
            rightOnesPattern = rightOnesPattern >> 2
            next = nextHigherOneBit | rightOnesPattern
        return next

File: result/translate/starchat/python2java/java/NEXT_HIGHER_NUMBER_WITH_SAME_NUMBER_OF_SET_BITS.java
public static int f_gold ( int x ) {
        int next = 0;
        if ( x!= 0 ) {
            int rightOne = x & -x;
            int nextHigherOneBit = x + rightOne;
            int rightOnesPattern = x ^ nextHigherOneBit;
            rightOnesPattern = rightOnesPattern / rightOne;
            rightOnesPattern = rightOnesPattern >> 2;
            next = nextHigherOneBit | rightOnesPattern;
        }
        return next;
    }